gateleen-user/src/main/java/org/swisspush/gateleen/user/UserProfileHandler.java

```diff
@@ -186,8 +170,13 @@ protected void cleanupUserProfile(final JsonObject profile, final Handler<JsonOb
         JsonObject profileCopy = profile.copy();
         Set<String> profileFieldNames = profileCopy.fieldNames();
         profileFieldNames.stream().filter(fieldName -> !userProfileConfiguration.isAllowedProfileProperty(fieldName)).forEach(fieldName -> {
-            log.debug("Removing property '{}' from user profile", fieldName);
-            profile.remove(fieldName);
+            if(userProfileConfiguration.isRemoveNotAllowedProfileProperties()) {
+                log.debug("Removing property '{}' from user profile", fieldName);
+                profile.put(fieldName, null);
```

Comment:
I assume there's a reason we do `put(fieldName, null)` in place of `remove(fieldName)`?

Reply:
Yes the reason is, that `remove(fieldName)` does not actually remove a field from storage when it's already there. This is because of the `merge=true` param that is used when the storage is called.